cloudinit/distros/mariner.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+# vi: ts=4 expandtab
+#
+# Copyright (C) 2021 VMware Inc.
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit import distros, helpers
+from cloudinit import log as logging
+from cloudinit import net, subp, util
+from cloudinit.distros import rhel_util as rhutil
+from cloudinit.settings import PER_INSTANCE
+
+LOG = logging.getLogger(__name__)
+
+NETWORK_FILE_HEADER = """\
+# This file is generated from information provided by the datasource. Changes
+# to it will not persist across an instance reboot. To disable cloud-init's
+# network configuration capabilities, write a file
+# /etc/cloud/cloud.cfg.d/99-disable-network-config.cfg with the following:
+# network: {config: disabled}
+"""
+
+
+class Distro(distros.Distro):
```

Comment:
Given that this class is almost entirely copied from the Photon distro class, why not just have the Distro class here inherit from `photon.Distro` and override the small pieces necessary?

Reply:
Totally make senses. I am working with my team to dig into this and find out what exactly are required for mariner.

Reply:
change as suggested to inherit from photon and override few methods